src/datamodel_code_generator/__init__.py

```
@@ -91,8 +92,24 @@ class DataclassArguments(TypedDict, total=False):
+
+    Uses LRU cache to avoid re-parsing the same YAML content,
+    which is especially useful when infer_input_type() and parse_raw()
+    parse the same content.
+    """
+    return yaml.load(text, Loader=SafeLoader)  # noqa: S506
+
+
 def load_yaml(stream: str | TextIO) -> YamlValue:
     """Load YAML content from a string or file-like object."""
+    if isinstance(stream, str):
+        import copy  # noqa: PLC0415
```

Ruff (0.14.10)

Unused `noqa` directive (non-enabled: `PLC0415`)

Remove unused `noqa` directive

(RUF100)